### ISSUE TITLE
How to increase state size?

### ISSUE THREAD

**Issue by richardburleigh** [saharNooby/rwkv.cpp#66]
First of all, amazing work on this implementation!

I wanted to really test pushing the context length to its limits (using the [chat example](https://github.com/saharNooby/rwkv.cpp/blob/master/rwkv/chat_with_bot.py)), but noticed that the state is always the same size when saved, even it had gone through 50k tokens worth of context.

How do I increase the size of the state available to accommodate the larger context?

Any help would be appreciated!

**Comment by york-yu-ctw** [comment]
If I understand correctly, the current implementation reusing state from previous inference, I am not sure how you are going to use the state, but storing state into binary file as the session snapshot when necessary may helps

**Comment by saharNooby** [comment]
By design, `state` tensor of RWKV has fixed size, no matter that you've processed 1 token or 100_000.

During training, RWKV is forced to use this fixed space to store relevant information as efficiently as possible. It is not perfect, of course; and may not work well outside of context lengths that were actually trained. As far as I know, there are only 8K ctx len fine tunes of RWKV.

To really increase size of `state` tensor, you would need to increase `n_embed` parameter of the model, and retrain it from scratch. Probably, it will improve memorization of relevant info.

TL;DR: RKWV has fixed state size by design, and nothing can be done about it without retraining from scratch.
